lib/index.js

```diff
@@ -456,65 +518,98 @@ setup_brush = function(graph){
         .attr("height",scatterHeight)
       .append("g")
         // this is just to move the picture down to the right margin length
-        .attr("transform", "translate(" + left + "," + top + ")");
+        .attr("transform", "translate(" + (marginForPCA.left) + "," + (marginForPCA.top) + ")");
 
     /*
     The zoom function works in the whole svg area.
     If we put 2 plots in one svg element, we will find that the circles in the scatter plot can reach the bar chart area when zooming in.
     So, we use one svg for scatter plot, one svg for bar chart to make sure that the zoom function only works in the scatter plot.
     */
     var svgForBar = d3.select(options.target).append("svg")
-        .attr("width", scatterWidth)
+        .attr("width", full_width - scatterWidth)
         .attr("height",options.barChartHeight)
       .append("g")
         // this is just to move the picture down to the right margin length
-        .attr("transform", "translate(" + left + "," + top + ")");
+        .attr("transform", "translate(" + (marginForBar.left) + "," + (marginForBar.top) + ")");
 
     graph.svgForPCA = svgForPCA;
     graph.svgForBar = svgForBar;
     return graph;
   };
 
 
+
   setup_graph = function(graph){
     //setup all the graph elements
     graph = setup_margins(graph);
     graph = setup_svg(graph);
     graph = setup_x_axis(graph);
     graph = setup_y_axis(graph);
 
-    /*
-    if you append your zoom/brush after your data points, the zoom/brush overlay will catch all of the pointer events,
-    and your tooltips will disappear on hover.
-    You want to append the zoom/brush before your data points, so that pointer events on the data generate a tooltip,
-    and those on the overlay generate a zoom/brush.
-    */
     var svg = graph.svgForPCA;
     var page_options = graph.page_options;
     var positonY = 0;
     var positonXForBrush = page_options.width - 80;
+    var options = graph.options;
 
-    svg.append("text")
-      .attr("y", positonY)
-      .attr("x", positonXForBrush)
-      .attr("class", "brushButton")
-      .attr("id", "notToggled")
-      .text("Toggle Brush")
-      .on("click", function(){
-        graph = setup_brush(graph);
-      });
-
+    /*
+      if you append your zoom/brush after your data points, the zoom/brush overlay will catch all of the pointer events,
+      and your tooltips will disappear on hover.
+      You want to append the zoom/brush before your data points, so that pointer events on the data generate a tooltip,
+      and those on the overlay generate a zoom/brush.
+    */
     graph = setup_zoom(graph);
-
     graph = setup_scatter(graph);
 
     graph = setup_legend(graph);
     graph = setup_x_axis_for_bar_chart(graph);
     graph = setup_y_axis_for_bar_chart(graph);
     graph = setup_bar(graph);
 
+    d3.select(".brushButton")
```

Comment:
I think that would be in the simple.js. But the call inside to the index.js would be a function that does this.